mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -232,6 +236,9 @@ impl Nexus {
         self.child_count -= 1;
         self.reconfigure(DREvent::ChildRemove).await;
 
+        // Update child status to remove this child
+        NexusChild::save_state_change();
```

Comment:
Should this be done before the child is removed or at least before we reconfigure the IO path?

Reply:
Done.